apps/barista-design-system/src/pages/single-page/components/toc/toc.ts

```diff
@@ -93,12 +87,17 @@ export class BaToc implements OnInit, AfterViewInit, OnDestroy {
      * and not to the current page. */
     ev.preventDefault();
     ev.stopImmediatePropagation();
-    const targetId = (ev.currentTarget as HTMLElement).getAttribute('href');
-    const target = this._document.querySelector(targetId || '');
+    const targetId = (ev.currentTarget as HTMLAnchorElement).hash;
+    const target = this._document.querySelector(targetId);
 
     if (this._platform.isBrowser && target) {
       // Has to be set manually because of preventDefault() above.
-      window.location.hash = targetId || '';
+      // We shouldn't use angular router since that would build the page again by setting its metadata again.
+      // By manipulating the url manually like below we avoid unnecassery page reloads.
+      const newLocation =
+        window.location.origin + window.location.pathname + targetId;
```

Comment:
Please inject the location and not access it via the window. Then the platform check should not be needed.

https://stackblitz.com/edit/angular-scroll-spy-routing?file=app%2Fhappy-scrolling.component.ts

Reply:
Instead of push in the new location and manually scrolling there, we can use the injected location and leverage `location.go`, that should work around the router problems we had and we can drop the platform checks.

Suggestion, while switching the element in the template away from a an `<a href="target"...` to something like: `span (click)="_handleToClick(target)"`

```ts
/** @internal handle the click on a TOC item */
  _handleTocClick(target: string): void {
    this._location.go(target);
  }

```
See: https://angular.io/api/common/Location#go